Tests: Counter::num_destroys’ is used uninitialized

I configured a debug release with `cmake -B Build/ladybird -DCMAKE_BUILD_TYPE=Debug`. However, building the tests fails:

```
$ Meta/ladybird.sh build
[0/2] Re-checking globbed directories...
[2634/2966] Building CXX object Lagom/Tests/AK/CMakeFiles/TestNeverDestroyed.dir/TestNeverDestroyed.cpp.o
FAILED: Lagom/Tests/AK/CMakeFiles/TestNeverDestroyed.dir/TestNeverDestroyed.cpp.o 
/usr/bin/c++ -DENABLE_COMPILETIME_FORMAT_CHECK -I/home/nathan/git/ladybird -I/home/nathan/git/ladybird/Userland/Services -I/home/nathan/git/ladybird/Userland/Libraries -I/home/nathan/git/ladybird/Build/ladybird/Lagom -I/home/nathan/git/ladybird/Build/ladybird/Lagom/Userland/Services -I/home/nathan/git/ladybird/Build/ladybird/Lagom/Userland/Libraries -I/home/nathan/git/ladybird/Meta/Lagom/../.. -I/home/nathan/git/ladybird/Meta/Lagom/../../Userland -I/home/nathan/git/ladybird/Meta/Lagom/../../Userland/Libraries -I/home/nathan/git/ladybird/Meta/Lagom/../../Userland/Services -I/home/nathan/git/ladybird/Build/ladybird -g -std=c++23 -fdiagnostics-color=always -Wall -Wextra -Wno-invalid-offsetof -Wno-unknown-warning-option -Wno-unused-command-line-argument -fno-exceptions -ffp-contract=off -Werror -Wno-expansion-to-defined -Wno-literal-suffix -Wno-dangling-reference -fno-semantic-interposition -fvisibility-inlines-hidden -Wno-maybe-uninitialized -Wno-shorten-64-to-32 -fsigned-char -ggnu-pubnames -fPIC -D_FILE_OFFSET_BITS=64 -ggdb3 -Og -MD -MT Lagom/Tests/AK/CMakeFiles/TestNeverDestroyed.dir/TestNeverDestroyed.cpp.o -MF Lagom/Tests/AK/CMakeFiles/TestNeverDestroyed.dir/TestNeverDestroyed.cpp.o.d -o Lagom/Tests/AK/CMakeFiles/TestNeverDestroyed.dir/TestNeverDestroyed.cpp.o -c /home/nathan/git/ladybird/Tests/AK/TestNeverDestroyed.cpp
In file included from /home/nathan/git/ladybird/Userland/Libraries/LibTest/TestCase.h:14,
                 from /home/nathan/git/ladybird/Tests/AK/TestNeverDestroyed.cpp:7:
/home/nathan/git/ladybird/Tests/AK/TestNeverDestroyed.cpp: In function ‘void should_not_destroy()’:
/home/nathan/git/ladybird/Tests/AK/TestNeverDestroyed.cpp:55:21: error: ‘*(Counter*)((char*)&n + offsetof(AK::NeverDestroyed<Counter>,AK::NeverDestroyed<Counter>::storage[0])).Counter::num_destroys’ is used uninitialized [-Werror=uninitialized]
   55 |     EXPECT_EQ(0, c->num_destroys);
/home/nathan/git/ladybird/Userland/Libraries/LibTest/Macros.h:40:21: note: in definition of macro ‘EXPECT_EQ’
   40 |         auto rhs = (b);                                                                                       \
      |                     ^
/home/nathan/git/ladybird/Tests/AK/TestNeverDestroyed.cpp:51:37: note: ‘n’ declared here
   51 |         AK::NeverDestroyed<Counter> n {};
      |                                     ^
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-shorten-64-to-32’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-unused-command-line-argument’ may have been intended to silence earlier diagnostics
cc1plus: note: unrecognized command-line option ‘-Wno-unknown-warning-option’ may have been intended to silence earlier diagnostics
cc1plus: all warnings being treated as errors
[2653/2966] Building CXX object Lagom/Userland/Libraries/LibWebView/CMakeFiles/LibWebView.dir/WebContentClient.cpp.o
ninja: build stopped: subcommand failed.
```

--- a/Tests/AK/TestNeverDestroyed.cpp
+++ b/Tests/AK/TestNeverDestroyed.cpp
@@ -44,20 +44,27 @@ TEST_CASE(should_construct_by_move)
     EXPECT_EQ(1, n->num_moves);
 }
 
-NO_SANITIZE_ADDRESS static void should_not_destroy()
-{
-    Counter* c = nullptr;
+struct DestructorChecker {
+    DestructorChecker(bool& destroyed)
+        : m_destroyed(destroyed)
     {
-        AK::NeverDestroyed<Counter> n {};
-        // note: explicit stack-use-after-scope
-        c = &n.get();
     }
-    EXPECT_EQ(0, c->num_destroys);
-}
+
+    ~DestructorChecker()
+    {
+        m_destroyed = true;
+    }
+
+    bool& m_destroyed;
+};
 
 TEST_CASE(should_not_destroy)
 {
-    should_not_destroy();
+    bool destroyed = false;
+    {
+        AK::NeverDestroyed<DestructorChecker> n(destroyed);
+    }
+    EXPECT(!destroyed);
 }
 
 TEST_CASE(should_provide_dereference_operator)

